src/target/drivers/display/spi/128x64x1_common.h

```diff
@@ -79,41 +57,27 @@ void LCD_Contrast(unsigned contrast)
 {
     //int data = 0x20 + contrast * 0xC / 10;
     LCD_Cmd(0x81);
-    int c = LCD_CONTRAST_FUNC(contrast);
-    LCD_Cmd(c);
+    if (HAS_OLED_DISPLAY) {
+        contrast = _oled_contrast_func(contrast);
+    } else {
+        contrast = _lcd_contrast_func(contrast);
+    }
+    LCD_Cmd(contrast);
 }
 
 void LCD_Init()
 {
-    //Initialization is mostly done in SPI Flash
-    //Setup CS as B.0 Data/Control = C.5
-    rcc_periph_clock_enable(get_rcc_from_pin(LCD_SPI.csn));
-    rcc_periph_clock_enable(get_rcc_from_pin(LCD_SPI_MODE));
-    GPIO_setup_output(LCD_SPI.csn, OTYPE_PUSHPULL);
-    GPIO_setup_output(LCD_SPI_MODE, OTYPE_PUSHPULL);
-    LCD_Cmd(0xE2);  //Reset
-    volatile int i = 0x8000;
-    while(i) i--;
-    lcd_display(0);     // Display Off
-    LCD_Cmd(0xA6);      // Normal display
-    LCD_Cmd(0xA4);      // All Points Normal
-    LCD_Cmd(0xA0);      // Set SEG Direction (Normal)
-    if (HAS_LCD_FLIPPED) {
-        LCD_Cmd(0xC8);  // Set COM Direction (Reversed)
-        LCD_Cmd(0xA2);  // Set The LCD Display Driver Voltage Bias Ratio (1/9)
+    lcd_init_ports();
+    if (HAS_OLED_DISPLAY) {
+        _oled_reset();
+        _oled_init();
     } else {
-        LCD_Cmd(0xEA);  // ??
-        LCD_Cmd(0xC4);  // Common Output Mode Scan Rate
+        _lcd_reset();
+        _lcd_init();
     }
-    LCD_Cmd(0x2C);      // Power Controller:Booster ON
-    i = 0x8000;
-    while(i) i--;
-    LCD_Cmd(0x2E);      // Power Controller: VReg ON
-    i = 0x8000;
-    while(i) i--;
-    LCD_Cmd(0x2F);      // Power Controller: VFollower ON
-    i = 0x8000;
+    volatile int i = 0x8000;
     while(i) i--;
```

Comment:
i suggest we change it to proper usleep

Reply:
i don't want to do that now.  we can have a separate pull-request.  Otherwise it is hard t debug failures.